components/tidb_query_datatype/src/codec/row/v2/compat_v1.rs

```diff
@@ -110,6 +110,9 @@ pub trait V1CompatibleEncoder: DatumFlagAndPayloadEncoder {
             FieldTypeTp::Null => {
                 self.write_u8(datum::NIL_FLAG)?;
             }
+            FieldTypeTp::Unspecified => {
```

Comment:
This change is not needed?

Reply:
There is no column info in `AnalyzeIndexReq`:
https://github.com/pingcap/tipb/blob/4fad48b4c8c3e17805a520d6c3a7655077ffca46/proto/analyze.proto#L36
Is this change for it?

Reply:
Yes, since we have already remove the dependency of `Analyze` to `BatchIndexScanner`